src/Microsoft.AspNetCore.OData/Routing/ODataActionSelector.cs

```diff
@@ -160,26 +255,96 @@ private bool TryMatch(IList<ParameterDescriptor> parameters, IList<string> avail
                     }
                 }
 
+                // if we can't find the parameter in the request, check whether
+                // there's a special model binder registered to handle it
+                if (ParameterHasRegisteredModelBinder(p))
+                {
+                    continue;
+                }
+
+                // if parameter is not bound to a key in the path,
+                // assume that it's bound to the request body
+                // only one parameter should be considered bound to the body
+                if (!matchedBody && RequestHasBody(context))
+                {
+                    matchedBody = true;
+                    continue;
+                }
+
                 return false;
             }
 
             return true;
         }
 
+        private bool RequestHasBody(RouteContext context)
+        {
+            string method = context.HttpContext.Request.Method.ToLowerInvariant();
```

Comment:
@habbes Is there any need in calling `.ToLowerInvariant()` here if you still have to supply the `StringComparison` comparison parameter in the line that follows?